cdisc_rules_engine/check_operators/helpers.py

```diff
@@ -185,15 +463,116 @@ def case_insensitive_is_in(value, values):
     return str(value).lower() in str(values).lower()
 
 
-def compare_dates(component, target, comparator, operator):
+def truncate_datetime_to_precision(date_string: str, precision: str):
+    dt = get_date(date_string)
+    if precision == "year":
+        return dt.replace(month=1, day=1, hour=0, minute=0, second=0, microsecond=0)
+    if precision == "month":
+        return dt.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+    if precision == "day":
+        return dt.replace(hour=0, minute=0, second=0, microsecond=0)
+    if precision == "hour":
+        return dt.replace(minute=0, second=0, microsecond=0)
+    if precision == "minute":
+        return dt.replace(second=0, microsecond=0)
+    if precision == "second":
+        return dt.replace(microsecond=0)
+    return dt
```

Comment:
Instead of if's, use a match statement here. The `precision` parameter's type should be passed as a DatePrecision member instead of str.

You could also use a dict of lambdas like you did for `_precision_check_functions`, but I think a match statement is slightly less verbose in both of these cases.